### DISCUSSION TITLE
Navigation_Beta: When Trying to show FullScreenCover Works fro Previews but not in Real Devices or Simulators

### DISCUSSION THREAD

**Discussion by BakrMohamedRBM** [pointfreeco/swift-composable-architecture#2125] [Beta]
```
struct AppMasterFeature: ReducerProtocol, NetworkHelper {
    @Preference(\.rootView) var rootView
    @Dependency(\.coreServices) var coreService

    struct State: Equatable {
        // MARK: - View states
        var rootView: AppRootView = .onBoarding
        var viewState: ViewState = .loaded
        
        //child
        @PresentationState var forceUpdate: ForceUpdateFeature.State?
    }
    
    enum Action: Equatable {
        case onAppear
        case checkForceUpdate
        case forceUpdateResponse(TaskResult<ForceUpdate>)
        case forceUpdate(PresentationAction<ForceUpdateFeature.Action>)
    }
    
    var body: some ReducerProtocolOf<Self> {
        Reduce { state, action in
            switch action {
            case .onAppear:
                state.rootView = rootView ?? .onBoarding
                return .none
            case .checkForceUpdate:
                guard isConnectedToInternet() else {
                    state.viewState = .offline(description: Str.youAreOffline.key)
                    return .none
                }
                return .task {
                    await .forceUpdateResponse(
                        TaskResult {
                            try await self.coreService.checkForceUpdate()
                        }
                    )
                }
            case .forceUpdateResponse(.success(let response)):
                state.forceUpdate = (response.updateRequired ?? false) ? ForceUpdateFeature.State() : nil
                return .none
            case .forceUpdateResponse(.failure(let error)):
                state.viewState = failHandler(error)
                return .none
            case .forceUpdate:
                  return  .none
            }
        }
        .ifLet(\.$forceUpdate, action: /Action.forceUpdate) {
            ForceUpdateFeature()
        }
    }
}
```

```
struct AppMasterView: View {
    let store: StoreOf<AppMasterFeature>

    var body: some View {
        WithViewStore(store, observe: { $0 }) { viewStore in
            LocalizedContentView{
                switch viewStore.rootView {
                case .onBoarding:
                    UnImplmentedView()
                case .welcome:
                    UnImplmentedView()
                case .home:
                    UnImplmentedView()
                }
            }
            .onAppear {
                viewStore.send(.onAppear)
                viewStore.send(.checkForceUpdate)
            }
            .sheet(store: store.scope(state: \.$forceUpdate, action: AppMasterFeature.Action.forceUpdate), content: ForceUpdateView.init(store:))
        }
    }
}

struct AppMasterView_Previews: PreviewProvider {
    static var previews: some View {
        LocalePreview{
            AppMasterView(store:
                    .init(
                        initialState: AppMasterFeature.State(),
                        reducer: AppMasterFeature()
                    )
            )
        }
    }
}
```

Full cover sheet showed perfectly on preview but nothing showed in simulators or real device

debug prints
```
✅ Did receive response 200 : for request /appVersion/updateRequired

Body:
 {
  "updateRequired" : true,
  "currentVersion" : "1.0.1",
  "minimumRequiredVersion" : "1.0.30"
}
received action:
  AppMasterFeature.Action.forceUpdateResponse(
    .success(
      ForceUpdate(
        updateRequired: true,
        currentVersion: "1.0.1",
        minimumRequiredVersion: "1.0.30"
      )
    )
  )
  AppMasterFeature.State(
    rootView: .onBoarding,
    viewState: .loaded,
-   _forceUpdate: nil
+   _forceUpdate: ForceUpdateFeature.State(viewState: .loaded)
  )
```

**Comment by BakrMohamedRBM** [comment]
after close looking i found

.ifLet(.$forceUpdate, action: /Action.forceUpdate) { ----} Is called before _forceUpdate: ForceUpdateFeature.State(viewState: .loaded) is updated

I don't know why

**Reply by mbrandonw** [reply]
Hi @BakrMohamedRBM, there isn't enough information here to diagnose the problem. Can you share a compiling project? Even better if you can reduce it to a very small repro.

**Reply by BakrMohamedRBM** [reply]
sure,  let me prepare one and share it with you

**Comment by danikkm** [comment]
Hi, I've encountered the same problem in my app. 
Imagine the following structure: `App.swift` -> `ContentView` -> `FeatureView`

**First example:**

The `App.swift` initializes the `ContentView` with the `AppFeature` store and `AppFeature` reducer. The `ContentView` contains the navigation stack that wraps the `FeatureView` (confirms to the reducer). Running the following structure causes issues with the presentation via `navigationDestination(store:_)` and `FeatureView` not drawing items in the `List`. 

**Second example:**
The `App.swift` initializes the wrapped `FeatureView` (confirms to the reducer) in the `NavigationStack`. Running the following structure works as expected. The presentation with `navigationDestination(store:_)` behaves accordingly, and `FeatureView` draws items in the `List`.

This issue is only reproducible while running the app on a simulator or physical device.

I've tried recreating the issue in the new project with the structure described above but couldn't reproduce the issue. Then I copy pasted the `FeatureView` with child views and reducers to my main project and was able to reproduce the issue (first example). I tried digging around and looking for what might cause the problem and found that removing `@Environment(\.scenePhase) var scenePhase` in the main project's `ContentView` fixes the abovementioned issues. But adding the `scenePhase` to the new project has no effect, and navigation works as expected. 🤷

**Reply by BakrMohamedRBM** [reply]
Thanks @danikkm Yes i was implemented also @Environment(\.scenePhase) var scenePhase and after removing it finally worked I spent hours trying to figure it out and I was not expecting to be caused of scenePhase

**Reply by BakrMohamedRBM** [reply]
@mbrandonw  Do you have any input why when adding  @Environment(.scenePhase) var scenePhase to app blocks navigation

**Reply by mbrandonw** [reply]
No idea. It would be best to share a small, simple project that reproduces the problem.

**Reply by BakrIOS91** [reply]
Here a issue demo project https://github.com/BakrIOS91/TCANavigation

**Reply by pyrtsa** [reply]
The reason that example breaks is because you've placed the `@Environment` property at the `App`'s level. So each time it changes the app's body is re-evaluated, probably recreating the store (`.init(...)`) etc. You should move the environment to be part of any `View` in your app such as a member of `ContentView`.
